### DISCUSSION TITLE
Do you use GitHub Discussions to suggest new ideas? - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by piers-sinclair** [SSWConsulting/SSW.Rules.Content#4880] [General]
Contributors: @wicksipedia @pierssinclairssw @SamBWagner @bradystroud @adamcogan 

Let's talk about this rule:
https://www.ssw.com.au/rules/use-github-discussions/

**Comment by GordonBeeming** [comment]
I think we should rather link to GitHub for the overall example of this working well. That then also lets people know where to go to suggest features for GitHub itself.

https://github.com/community/community/discussions

**Comment by billyterblanche** [comment]
I haven't used GitHub Discussions yet, but from what I can see it seems to be pretty neat. I quite like the upvote feature as this allows for better visibility to the topics that matter.

**Comment by tiagov8** [comment]
I've experienced a circumstance where I felt my new Issue wasn't in the right place... because it was more of a question than a request. Now I know where it belonged to :)

PS. Much better than an email thread!

**Comment by 2fernandez** [comment]
I like it. First time I used GitHub discussions was on @matt-goldman blog. 
Very cool Tech. Though I though find that GitHub's general UX a bit lacking.

**Comment by leehawkinsssw** [comment]
I wasn't aware of this feature in GitHub but it looks useful and has all the features I'd expect of a feature of this type. I can see plenty of use cases for this and maybe we can handle the "Suggestions" items in SugarLearning using GitHub Discussions rather than emails.

I'd like us to decide whether GitHub Discussions is a thing or are a thing, and then be consistent in the rule and other places we refer to it/them.

**Comment by bradystroud** [comment]
I think there are some frequent emails that we could move to Discussions to reduce inbox noise

**Comment by brydeno** [comment]
This certainly makes sense when used as part of the whole github package. ie keeping all your concerns handled on one platform. Makes a lot less sense for houses using say Jira to track work.

**Comment by jakebayliss** [comment]
Haven't had the chance to use it yet but this looks great and packs a whole lot of cool features. We will definitely be able to move a whole lot of discussions (pun intended) away from email/teams and onto here and have them available to everyone

**Comment by BrookJeynes** [comment]
GitHub discussions seems like a great idea for projects that are based on Github. It allows everything related to the project to be kept on a single platform allowing easier access to the required information.

**Comment by UlyssesMaclaren** [comment]
Very cool... I always wondered what the correct place to store these "pre-PBI" discussions should be... up till now it's been Teams chats and emails

**Comment by Levijj22** [comment]
Seems like a great platform, especially the upvote feature! 
Less emails the better!

**Comment by danielmackay** [comment]
I love the idea of GitHub discussions.  It makes it much easier to follow a conversation especially if you get 'looped' in halfway through.  I do wonder about privacy though.  Some discussions may not be best had in public, and some people may not be as open/honest if they know their comments can be seen by everyone.

However, I think in a lot of cases it can be great.  Often good ideas can feed off other good ideas and iterate to become amazing ideas!
	
Love the idea of converting discussions to issues, to instantly include all context for a PBI.

As for the rule, it provides some great general advise as to how discussions should be used.  I would like to see some proposals as to how specifically we would use this at SSW.

**Comment by tkapa** [comment]
I like the idea of primarily using GitHub Discussions when it comes to GitHub projects. There seems to be a lot of nice to have features from the get go, which makes them valuable.

I think that the rule is far too long. 

The dot points at the beginning get across the major uses for discussions which is great, but then goes back through them again, which isn't necessary. 

The flow of Discussion -> Issue -> Pull Request. I don't believe that the rule needs more content past this.

**Reply by SamBWagner** [reply]
@tkapa test

**Comment by kikibianc** [comment]
Discussions seems cool, but I doubt the SysAdmin team will use it a lot :)

**Comment by ozdemir-mehmet** [comment]
Looks fine, but I'll probably stick with Issues

**Comment by matt-goldman** [comment]
Discussions are great, especially now that you can convert to an issue.

As an aside, not sure about the 'cutting down on email' noise part. I've received a notification email for every comment in this discussion before I've even looked at it (as I was tagged in an earlier comment). Not a complaint, just an observation. Of course you can easily unsubscribe from email notifcations.

The reasons Discussions are superior to email are because (in order):
* You can see the whole conversation in one place
* No need to worry about conversation fragmentation in your inbox and searching through different chains
* No need to worry about replying to an outdated email, etc...
* You can upvote and react to comments
* Comment can be organised into threads, but you can still see the whole conversation in one place

**Reply by adamcogan** [reply]
It's "cutting down on *sending* emails" ... not receiving emails 😂 

Getting a notification from GitHub or Facebook 'your thread has been updated' is useful for some, not for others.

-a

**Comment by leahy268** [comment]
I have used discussions to raise the case for features in products that I am using. 

Recently when I did this the owner of the repo shortly afterwards provided the feedback that this was probably not very difficult for him to implement and promptly changed it into an issue.  He showed me samples of the output and allowed me to give him feedback.  Other people also joined the discussion and told him how useful it would be and gave their feedback.

He then created a pull request and it is now in the new version.  

This is so much better than an email which is only seen by the people included in the email.

**Comment by TobyChurches** [comment]
Although I haven't used this feature previously, It looks great! I think it makes sense to keep the discussions and issues in one spot, rather than spreading them across multiple platforms. It's definitely a bonus that the discussions can be made public, allowing new devs to easily gain context on the subject.

I will definitely be using this feature 😁

**Comment by AntPolkanov** [comment]
Nice feature as it allows to keep Code, Issues and Discussions close together! 

One question though. As many people wanna use Discussions instead of email, should a discussion be subject to the Checked by XXX rule (https://www.ssw.com.au/rules/checked-by-xxx/) ?

**Reply by SamBWagner** [reply]
I think a Discussion is the part that leads to a 'checked by' in an Issue. If Discussions are hindered by roadblocks like that then they'll lose a lot of value.

**Comment by cvionnet** [comment]
I like the way it can be used for product communication and community interaction (Poll, Product announcements, upvotes), also the integration with other GitHub features. A great tool for projects on GitHub.

As the number of source of discussion increases (Discussions, emails, Teams), it needs to be clear which tool is used for the project to avoid any miss in the flow of communication.

**Comment by jontygardner** [comment]
I love it provides a centralized platform for team discussions, reducing the need for using emails.

A con is the customization options - working with video a lot we need to show visuals easy e.g. videos, gifs etc.

**Comment by lukecookssw** [comment]
Discussions would need to be integrated into everyday workflow (requires discipline) or it will end up as a black hole of abandoned comments and/or "almost" issues.

**Reply by SamBWagner** [reply]
100%. A transition to using it will require some fairly fundamental changes to the "SSW" etiquette and standards. 

It's likely worth it though imo. Reading all of these comments was SO much easier then a long email chain.

**Comment by SamBWagner** [comment]
I think Discussions are a great tool and are sorely underutilized by SSW. 

It will take a bit off effort to transition from our other mediums to using these but the value at the end, I believe, is worth it

Rule is WAY too long though and it doesn't need to explain **what** a Discussion is. Just how we use it. 

I do like the little list of what they're used for as well 😈 (I'm a hypocrite and I don't care haha)

Note: These update in real-time. Which is hectic

**Comment by samsmithssw** [comment]
The upvote feature is pretty cool, allows comments to be prioritised by value (so when there are heaps of comments like in this discussion you know which ones are the most important to read first). Readability is a lot better than an email thread.

**Comment by amankumarrr** [comment]
I wasn't aware of this cool feature, it would be really handy in open-source projects where we can easily discuss feedback, issues, and maybe new ideas and track them, instead of following up on the emails.

**Comment by KahaMason** [comment]
I started using github discussions on a newer client project and its definitely helped boost collaboration and crowd sourcing ideas from the various teams.

**Comment by antonvojkovicssw** [comment]
I don't use GitHub that much (power platform dev 🙃), but I think it's a pretty cool feature. I do think for "crazy idea" of PBI would be better to chat about it in a teams call, since a GitHub discussion can drag the decision a lot.

**Comment by zacharykeeping** [comment]
I like this - seems like a good central place to keep general discussions that might not necessarily be issues, but could also easily be converted to one.

**Comment by AdriT9** [comment]
Good feature that can be a source of information for someone that doesn't understand much about this world (like myself) 😅

**Comment by christoment** [comment]
I like the idea of having a discussion platform which can take care of questions in a GitHub issue 🙂

**Comment by steven0x51** [comment]
I haven't used discussion yet, but it provides a centralized place where you can find and post all the repo related issues, and reducing the need for using emails.

**Comment by NadeekaJK** [comment]
Haven't used this before, would be cool to post general suggestions and Sugar Learning suggestions.

**Comment by piers-sinclair** [comment]
GitHub Discussions look cool, but I am still not 100% convinced that it can be organically incorporated into an organization or project with ease.

**Comment by Marxoz** [comment]
Github discussions is a good feature to add to your projects. I just wonder if it can get a bit messy with too many discussions and how that will be controlled. A group is one option of course, but there might be 100s of discussions in a single group. I will probably have an answer when I start using it more.....

**Comment by Geordie88** [comment]
I like the idea in concept as it could help cut down on long messy email threads but I have a couple of concerns. 

On a UX level I don't like the post comment field being at the bottom of the page (I don't know if there is a setting to tweak for this). 

More broadly depending on the size of the team and the number of issues being discussed I can see it getting out of hand and it is hard to find the issue you are looking for. That said I am new to GitHub discussions so there may be ways to manage this.

**Comment by apforsyth** [comment]
I was interested to learn about the advantages of Github Discussions and wonder if done videos can be linked there as well - e.g.https://youtu.be/CjwkGk1LcJ0

The rule mentions areas for improvement but doesn't address criticisms I found via Chat GPT (validity unknown!).

My own suggestion is to be able to comment on the thread without having to scroll to the bottom of all the comments to find the "comment" button might be good.

From Chat GPT:
GitHub Discussions has received some criticisms, including:

Limited moderation tools: Some users have reported that the moderation tools available in GitHub Discussions are limited, making it difficult to manage discussions and maintain a positive environment.

Search functionality: Some users have reported that the search functionality in GitHub Discussions is not always accurate or effective, making it difficult to find relevant information and discussions.

Learning curve: GitHub Discussions is built into the GitHub platform, and some users may find the interface and functionality confusing or difficult to use, particularly if they are not already familiar with GitHub.

Lack of customization: GitHub Discussions has a limited set of customization options, making it difficult for projects to create a unique and branded experience for their users.

Inadequate integration with GitHub issues: Some users have reported that GitHub Discussions are not well integrated with GitHub issues, making it difficult to track discussions and issues in a single place.

These criticisms may vary in importance depending on the specific needs and goals of a project, and some of these issues may be addressed in future updates to the platform. Nevertheless, they are factors that should be considered when deciding whether to use GitHub Discussions for a given project.

**Comment by stephanfakossw** [comment]
Looks great and reduces the need for emails!

**Comment by Harry-Ross** [comment]
I thought it was great, it makes it way easier to provide feedback/suggestions to a project without creating a GitHub issue, which I believe will allow for a more accessible discussion medium. It could also allow for less email/Teams communication.

**Comment by JackDevAU** [comment]
GitHub Discussions are an awesome place to discuss rules, CtF's as well as a great place to discuss and gather feedback from everyone at SSW. Super excited to see less emails and move discussions over here!

**Comment by chrisschultzssw** [comment]
GitHub discussions looks like a great place for... discussions! I like that they link through to the Issue and then a PR - when done correctly, there will be no more searching through emails to find history.

Another +1 to less emails!

**Comment by jimmy1992abc** [comment]
GitHub Discussion offers a unique platform for sharing and developing ideas. It provides a convenient space for conversing and obtaining feedback, helping to ensure that good ideas don't go to waste due to time constraints or lack of feedback.

This feature is not just a new addition to the internet, but a new aspect of GitHub. We can use it as a platform that can be utilized to transport ideas before drafting a formal email.

For example, if I have an idea, I need to have some conversations with others and then draft an email in seriousness. Those ideas might be gone from further development for some reason. Some of them could be like lack of time, no feedback from others, etc.  I believe ideas should be more welcome and not limited by the procedures and time sacrificed.

GitHub Discussion might be a great opportunity to keep those ideas and make the conversation easier, we should encourage people to post any ideas and everyone can have feedback on any topic. Those discussions could be casual and open, it allows for diverse perspectives from individuals in different roles and cultural backgrounds to contribute to the discussion.

 I hope GitHub Discussion could be our internal chat board. By encouraging people to post their ideas, we can foster a culture of open communication and generate valuable feedback for the company.

**Comment by PennyWalker** [comment]
This is something totally new to me! I love the fact it has levels of escalation, so if you're not sure how to proceed you can start a discussion. I think it's a great place to keep a record of any issues or questions that arise for a project. Email do not allow this level of tracking, love it.

**Comment by KostaMadorsky** [comment]
Love it.
It definitely was a missing step before creating an issue.
The only thing is that in public repos if you open a discussion for a new feature it may get lower priority compared to issues.

**Comment by jaydenalchin** [comment]
GitHub Discussions are new to me, but seem like a great way to keep discussions that don't relate to a specific PBI in a place that is still solidly attached to the relevant project.

**Comment by bahjatalaadelssw** [comment]
I'll try this out for when I work on editing a new SSW Rule. This would have been using for the video editing terms rules!

**Comment by drwharris** [comment]
I like it. I can see some great uses for PO's

**Comment by wicksipedia** [comment]
I'm a fan of using GitHub discussions to discuss/make decisions as opposed to email - primarily because email is a data island. IMHO this is a cleaner way of organizing the conversation.

**Comment by AttackOnMorty** [comment]
Awesome feature! It creates a buffer zone in the workflow to consolidate ideas.

**Comment by jimmidier** [comment]
Quite an ideal alternative for emails. We should have used it much ealier back when it was first released in 2020!

**Comment by sylhuang** [comment]
This feature is super convenient. I prefer this to group emails so that some less urgent, more casual brainstorming could happen without bombarding everyone's mailbox.

**Comment by jerryluossw** [comment]
It's a great feature to discuss somethings instead of emails, the email only needs to contain a link to the discussion and the result

**Comment by KristenHu** [comment]
Like this idea, compared with email, it helps us to separate these disccussion and quetions to their different repos.

**Comment by johnxu16** [comment]
I prefer using GitHub discussion. It is much easier for us to connect other resources (ex. repo and scrum board) to a GitHub discussion.

**Comment by AnkyChe** [comment]
I like it. Compared to emails. GitHub discussion makes it easier to discuss and record. And we can share wonderful ideas together.

**Comment by hveraus** [comment]
I think github discussions should replace stackoverflow, devs will be much more benefited.

**Comment by SofieHong** [comment]
Great idea! Compared to emails, GitHub discussions are a convenient way to see what other people say.

**Comment by premradhakrishnan** [comment]
Pros - 
 - It fills the gap that exists between ideas floating around and GitHub Issues. 
 - The upvote feature is great as popular ideas get more visibility
 - Also provides transparency to how long ago a feature was requsted.

Cons -
- On large projects, I think a certain amount of admin moderation is required to keep the number of discussions in check or it can become like StackOverflow where the same question gets posted multiple times.

**Comment by vladislav-kir** [comment]
I never used GitHub Discussions before, but I think it can be useful for some projects. For example, putting suggestions for SugarLearning or Rules in Discussions would be better than sending them in emails.

**Comment by Berber-Mama** [comment]
Wow, it looks like a new world to me! It's good to see others' comments here, it helps me to understand what's going on and the discussion history. 😊

**Comment by Freego1783** [comment]
I never had the occasion to use GitHub discussions, but it looks promising. I think it can be really useful to manage customer support/FAQ and product announcements.

**Comment by AlexRothwell** [comment]
I'm loving the looks of GitHub Discussions so far, it absolutely looks like a great tool for discussing new feature ideas before they have been fully worked out

Also, being a central source of truth means that everyone can always see the latest state of things

Additional visibility to all who can access the repository is another bonus - someone might not have been included in initial discussions but have something to add. Rather than relying on everyone in an email thread to know of everyone who's interested, people can add themselves to the discussion

**Comment by camillars** [comment]
Great feature for collaboration from GitHub - cool to see everyone's thoughts together and be able to engage with each one.

**Comment by Hona** [comment]
Its good for a more casual forum setting from GitHub. However, it is now hard to differentiate between issue/discussions. I feel it needs some wider adoption to make the process smooth.
We could use Discussions as a knowledge base, or even a decision record (with the reasoning, from the conversation!). 

Also, we can use this automatically linked to Rules for example via: https://giscus.app/

**Comment by ozair-ashfaque** [comment]
Awesone feature, we can replace  emails with GitHub discussions

**Comment by Ben0189** [comment]
This is the first time I have used GitHub discussion, and it has better readability than email. It is a great idea to replace some emails with GitHub discussion

**Comment by AshAnil** [comment]
The GitHub discussion is an excellent feature... Reduces the email noise, and better chat history. I loved the up votes option but would prefer the names of the person who voted in that particular discussion

**Comment by JackDuan123** [comment]
What I learned:

All companies are trying to do everything on their platform, it seems like they are under huge pressure from their share holders.
I feel sorry for them.

**Comment by calumjs** [comment]
I learnt that this feature exists

**Comment by ravidhaduk** [comment]
Knew about this but never used it. Seems like fewer emails from now on (if we really use it 😊). 👍👍

**Comment by Calvin-Lieu** [comment]
Haven't used this before, looks like a convenient feature which is worth looking into.
